mule-extensions-api/src/main/java/org/mule/runtime/extension/api/introspection/AbstractNamedImmutableModel.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.introspection;
+
+import java.util.Set;
+
+import org.apache.commons.lang3.builder.ToStringBuilder;
+
+
+/**
+ * Base class for immutable implementations of the introspection model
+ *
+ * @since 1.0
+ */
+public abstract class AbstractNamedImmutableModel extends AbstractImmutableModel implements Named
+{
+
+    private final String name;
+
+    /**
+     * Creates a new instance
+     *
+     * @param name            the model's name
+     * @param description     the model's description
+     * @param modelProperties A {@link Set} of custom properties which extend this model
+     * @throws IllegalArgumentException if {@code name} is blank
+     */
+    protected AbstractNamedImmutableModel(String name, String description, Set<ModelProperty> modelProperties)
+    {
+        super(description, modelProperties);
```

Comment:
All immutable models extend from `AbstractImmutableModel` inheriting implementations for `Described, EnrichableModel`. 
Most models also implement `Named`, so this `AbstractNamedImmutableModel` provides a common implementation for all the models that are Named as well as Described and EnrichableModel.

This in inheritance exists to provide a way for an ImmutableModel to be unnamed, since it makes no sense to set a name for an `OutputModel`